File: credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    /* get input */
    long long cc;
    do
    {
        printf("Card number to verify: ");
        cc = get_long_long();
    }
    while (cc < 0);
    
    /* define variables */
    long long checker = cc;
    int checkSum = 0;
    int digit;
    
    /* starting w/ 2nd to last number, multiply every other # by two and add it to checksum */
    do
    {
        /* cut off one digit, grab last digit, multiply it by two */
        checker = checker / 10;
        /*printf("checker is: %lld\n",checker);*/
        digit = (checker % 10) * 2;
        
        /* if the digit is over two, add each of its digits to the checksum, otherwise add the digit to checksum */
        if (digit > 9)
        {
            do
            {
                int x = digit % 10;
                checkSum += x;
                digit = digit / 10;
            }
            while (digit > 0);
        }
        else
        {
            checkSum += digit;
        }
        
        /* cut off one number */
        checker = checker / 10;
        /*printf("checker now is: %lld\n",checker); */
    }
    while (checker >= 1);
    
    /* starting at last digit in cc, add up every other digit */
    checker = cc;
    do
    {
        digit = checker % 10;
        checkSum += digit;
        checker = checker / 100;
    }
    while (checker >= 1);
    
    /* if last digit in checksum is 0 and card # is long enough/recognized it's valid */
    if (checkSum % 10 == 0 && cc > 1000000000000)
    {
        
        /* get first and second digits of cc # */
        do
        {
            cc = cc / 10;
        }
        while (cc > 100);
        
        int second = cc % 10;
        int first = cc / 10;
        int sum = first + second;
        
        /* check if VISA */
        if (first == 4) {
            printf("VISA\n");
        }
        /* check if AMEX */
        else if (first == 3 && (sum == 7 || sum == 10))
        {
            printf("AMEX\n");
        }
        else if (first == 5 && ((5 < sum) && (sum < 11)))
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");    
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: credit.py


def firstCheck(cc):
    result = 0
    
    while(cc >= 1):
        
        # cut off one digit of card number and multiply by 2
        cc = cc // 10
        digit = (cc % 10) * 2
        
        # if it is over two digits long add each digit together
        if (digit > 9):
            result += digit % 10
            digit = digit // 10
        
        result += digit
        cc = cc // 10
    
    return result

def secondCheck(cc):
    result = 0
    
    while (cc >= 1):
        digit = cc % 10
        result += digit
        cc = cc // 100
        
    return result
    
def isValid(cc):
    
    firstResult = firstCheck(cc)
    secondResult = secondCheck(cc)
    checkSum = firstResult + secondResult
    
    if (checkSum % 10 == 0):
        return True
    else:
        return False
    
while True:
    try:
        cc = int(input("What's your card number (you can trust me): "))
    except ValueError:
        print("I dont think that is a valid number. No spaces or any special characters plz")
        continue
    if (isValid(cc) is True and cc > 1000000000000):
        
        ccStr = str(cc)
        firstDigit = int(ccStr[0])
        secondDigit = int(ccStr[1])
        sum = firstDigit + secondDigit
        
        if (firstDigit is 4):
            cardType = "Visa"
        elif (firstDigit is 3 and (sum is 7 or sum is 10)):
            cardType = "Amex"
        elif (firstDigit is 5 and ((5 < sum) and (sum > 11))):
            cardType = "Mastercard"
        else:
            print("That is not a valid credit card type. You may have been bamboozled or you are trying to bamboozle me. Not gonna happen!")
            break
        
        print(f"Sweet {cardType} you got there fam")
        break
    else:
        print("That is not a valid credit card. You may have been bamboozled or you are trying to bamboozle me. Not gonna happen!")
        break
